src/Core/Components/DataGrid/FluentDataGrid.razor

```diff
@@ -5,260 +5,313 @@
-                }
-            </FluentDataGridCell>
-        </FluentDataGridRow>
+				}
+				else
+				{
+					@EmptyContent
+				}
+			</FluentDataGridCell>
+		</FluentDataGridRow>
+
+	}
+
+	private void RenderLoadingContent(RenderTreeBuilder __builder)
+	{
+		string? style = null;
+		string? colspan = null;
+		if (DisplayMode == DataGridDisplayMode.Grid)
+		{
+			style = $"grid-column: 1 / {_columns.Count + 1}";
+		}
+		else
+		{
+			colspan = _columns.Count.ToString(CultureInfo.InvariantCulture);
+		}
+
+		<FluentDataGridRow Class="@LOADING_CONTENT_ROW_CLASS" TGridItem="TGridItem">
+			<FluentDataGridCell Class="loading-content-cell" Style="@style" colspan="@colspan">
+				@if (LoadingContent is null)
+				{
+					<FluentStack HorizontalGap="3" HorizontalAlignment="@HorizontalAlignment.Center">
+						<FluentSpinner Size="SpinnerSize.Medium" /> <div>Loading...</div>
+					</FluentStack>
+				}
+				else
+				{
+					@LoadingContent
+				}
+			</FluentDataGridCell>
+		</FluentDataGridRow>
+	}
+
+	[ExcludeFromCodeCoverage(Justification = "This method requires a failing db connection and is to complex to be tested with bUnit.")]
+	private void RenderErrorContent(RenderTreeBuilder __builder)
+	{
+		if (_lastError == null)
+		{
+			return;
+		}
 
-    }
+		string? style = null;
+		string? colspan = null;
+		if (DisplayMode == DataGridDisplayMode.Grid)
+		{
+			style = $"grid-column: 1 / {_columns.Count + 1}";
+		}
+		else
+		{
+			colspan = _columns.Count.ToString();
+		}
 
-    private void RenderLoadingContent(RenderTreeBuilder __builder)
-    {
-        string? style = null;
-        string? colspan = null;
-        if (DisplayMode == DataGridDisplayMode.Grid)
-        {
-            style = $"grid-column: 1 / {_columns.Count + 1}";
-        }
-        else
-        {
-            colspan = _columns.Count.ToString(CultureInfo.InvariantCulture);
-        }
+		<FluentDataGridRow Class="@ERROR_CONTENT_ROW_CLASS" TGridItem="TGridItem">
+			<FluentDataGridCell Class="empty-content-cell" Style="@style" colspan="@colspan">
+				RenderActualError(_lastError)
+			</FluentDataGridCell>
+		</FluentDataGridRow>
+	}
 
-        <FluentDataGridRow Class="@LOADING_CONTENT_ROW_CLASS" TGridItem="TGridItem">
-            <FluentDataGridCell Class="loading-content-cell" Style="@style" colspan="@colspan">
-                @if (LoadingContent is null)
-                {
-                    <FluentStack HorizontalGap="3" HorizontalAlignment="@HorizontalAlignment.Center">
-                        <FluentSpinner Size="SpinnerSize.Medium" /> <div>Loading...</div>
-                    </FluentStack>
-                }
-                else
-                {
-                    @LoadingContent
-                }
-            </FluentDataGridCell>
-        </FluentDataGridRow>
-    }
+	[ExcludeFromCodeCoverage(Justification = "Circumvent EFCC shortcomings when using razor.")]
+	private void RenderActualError(RenderTreeBuilder __builder, Exception ex)
```

Comment:
To avoid using `ExcludeFromCodeCoverage`, you should move the code to the .cs file using a code like this.
Can you do that in a new PR to increase the code coverage and to have a "best clean code" ? :-)

```csharp
private void RenderActualError(RenderTreeBuilder builder)
{
    if (ErrorContent is null)
    {
        builder.AddContent(0, "An error occurred while retrieving data.");
    }
    else
    {
        builder.AddContent(1, ErrorContent(_lastError));
    }
}
```